File: kq-projection-dotnet/kq-projection/Common/Models/Keys/ProjectionKey.cs
using KQ_Projection.Common.Enums;
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Keys;

public class ProjectionKey
{
    [JsonPropertyName("playerId")]
    public long playerId { get; set; }

    [JsonPropertyName("gameId")]
    public Guid gameId { get; set; }

    [JsonPropertyName("eventType")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public EventType eventType { get; set; }

    public static ProjectionKey Create(long playerId, Guid gameId, EventType eventType) { 
        return new ProjectionKey {  playerId = playerId, gameId = gameId, eventType = eventType };
    }
}

File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/kafkakeys/ProjectionKey.java
package com.kafkaquest.kq.common.util.models.kafkakeys;

import java.util.UUID;

import com.kafkaquest.kq.common.util.models.events.enums.EventType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProjectionKey {
    private Long playerId;
    private UUID gameId;
    private EventType eventType;
}
